client/rest-high-level/src/main/java/org/elasticsearch/client/core/TermVectorsResponse.java

```diff
@@ -0,0 +1,452 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.core;
+
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Comparator;
+import java.util.Objects;
+
+public class TermVectorsResponse extends ActionResponse implements ToXContentObject {
+    private final String index;
+    private final String type;
+    private final String id;
+    private final long docVersion;
+    private final boolean found;
+    private final long tookInMillis;
+    private List<TermVector> termVectorList = null;
```

Comment:
It feels a little weird to have just this part be mutable.

Reply:
is it possible for the `ConstructingObjectParser` to pass in a `NamedObjects` `constructorArg`? Right now on line 67 is where its being mutated.